packages/server/lib/controllers/connection.controller.ts

```diff
@@ -116,10 +118,12 @@ class ConnectionController {
 
             const template: ProviderTemplate | undefined = configService.getTemplate(config.provider);
 
+            // if instantRefresh is true, we already refreshed the credentials
             if (
-                connection.credentials.type === ProviderAuthModes.OAuth2 ||
-                connection.credentials.type === ProviderAuthModes.App ||
-                connection.credentials.type === ProviderAuthModes.OAuth2CC
+                !instantRefresh &&
+                (connection.credentials.type === ProviderAuthModes.OAuth2 ||
+                    connection.credentials.type === ProviderAuthModes.App ||
+                    connection.credentials.type === ProviderAuthModes.OAuth2CC)
             ) {
                 const {
                     success,
```

Comment:
It's quiet confusing to have a refresh in getConnectionCredentials and one here, and I feel like this call is already handled in connection.service.ts. This is the exact same condition but maybe I'm missing something